mbuild/compound.py

```diff
@@ -1457,6 +1442,42 @@ def is_independent(self):
                         return False
             return True
 
+    def check_for_overlap(self, excluded_bond_depth, minimum_distance=0.10):
```

Comment:
Add a check for minimum_distance against the box length of the compound. Throw an error if larger than 1/2 the box length.